library/src/main/java/org/xmtp/android/library/Conversations.kt

```diff
@@ -175,8 +175,10 @@ data class Conversations(
             }
         }
 
-        conversationsByTopic += newConversations.filter { it.peerAddress != client.address }
-            .map { Pair(it.topic, it) }
+        val regex = Regex("[A-Za-z0-9\\-\\/]+")
```

Comment:
So I dug a little bit deeper on this and the conversation topic name is a 32-byte random alphanumeric string, such as `m-AhUFyewKxM1pxr1JmJJkDK3tDn3gQ0Op` We [wrap](https://github.com/xmtp/xmtp-android/blob/main/library/src/main/java/org/xmtp/android/library/messages/Topic.kt#L29) the topic with the xmtp identifying information so that's highly controlled and should never have garbage characters. You can see a [PR](https://github.com/xmtp/xmtp-js/pull/385) that fixed the underlying issue in JS and why we have to filter now because of random conversations that got corrupted. I think if we are going to match this work the Regex would be `/^[\x00-\x7F]+$/`. But this feels like the correct spot.

It also feels worth it to write a [test](https://github.com/xmtp/xmtp-android/blob/main/library/src/androidTest/java/org/xmtp/android/library/ConversationsTest.kt) to validate that this works. Might be hard to create a conversation with a corrupt topic at this point but you could try and do some manual overrides to confirm and not merge the test if that makes sense.